packages/fetchai/skills/carpark_client/behaviours.py

```diff
@@ -19,11 +19,99 @@
 
 """This package contains the behaviours of the agent."""
 
+from typing import Any, cast
+
+from aea.skills.behaviours import TickerBehaviour
+
+from packages.fetchai.protocols.fipa.message import FipaMessage
+from packages.fetchai.skills.carpark_client.strategy import Strategy
 from packages.fetchai.skills.generic_buyer.behaviours import (
     GenericSearchBehaviour,
     GenericTransactionBehaviour,
 )
+from packages.fetchai.skills.generic_buyer.dialogues import FipaDialogue, FipaDialogues
 
 
 SearchBehaviour = GenericSearchBehaviour
 TransactionBehaviour = GenericTransactionBehaviour
+
+
+class ProposalCheckBehaviour(TickerBehaviour):
+    """
+    Controls the timer to check if proposals have been received.
+
+    :param TickerBehaviour: generic ticker behaviour to be inherited from
+    """
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the proposal check behaviour."""
+        super().__init__(**kwargs)
+        self.counter = 0
+
+    def setup(self) -> None:
+        """Set up the behaviour."""
+
+    def act(self) -> None:
+        """Act according to the behaviour."""
+        strategy = cast(Strategy, self.context.strategy)
+        if not strategy.waiting_for_proposals:
+            return
+
+        # check if every agent has sent at least one proposal
+        if set(strategy.sent_proposals).issubset(
```

Comment:
This just checks you have received some proposals. You shouldn't necessarily trigger decision making here. Only when the timeout is reached.